docker/data-science/notebooks/solr_collection.py

```diff
@@ -17,53 +15,18 @@ def commit(self):
         return requests.post(f"{SOLR_URL}/{self.name}/update?commit=true")
         time.sleep(10) # temporary
 
-    def write_from_csv(self, file, more_opts=False):
-        print(f"Loading {self.name}")
-        spark = SparkSession.builder.appName("AIPS").getOrCreate()
-        reader = spark.read.format("csv").option("header", "true").option("inferSchema", "true")
-        if more_opts:
-            reader = reader.option("charset", "utf-8").option("quote", "\"").option("escape", "\"").option("multiLine","true").option("delimiter", ",")
-        csv_df = reader.load(file)
-        if more_opts and "category" in more_opts:
-            # We can rely on automatic generation of IDs, or we can create them ourselves. 
-            # If we do it, comment out previous line
-            # .withColumn("id", concat(col("category"), lit("_") col("id")))
-            csv_df = csv_df.withColumn("category", lit(more_opts.get("category"))).drop("id")
-        print(f"{self.name} Schema: ")
-        csv_df.printSchema()
-        options = {"zkhost": AIPS_ZK_HOST, "collection": self.name,
-                   "gen_uniq_key": "true", "commit_within": "5000"}
-        csv_df.write.format("solr").options(**options).mode("overwrite").save()
-        self.commit()
-        print("Status: Success")
-        self.commit()
-    
-    def write_from_dataframe(self, dataframe):
+    def write(self, dataframe):
         opts = {"zkhost": AIPS_ZK_HOST, "collection": self.name,
                 "gen_uniq_key": "true", "commit_within": "5000"}
         dataframe.write.format("solr").options(**opts).mode("overwrite").save()
         self.commit()
     
-    def write_from_sql(self, query, spark=None):
-        if not spark:
-            spark = SparkSession.builder.appName("AIPS").getOrCreate()
-        opts = {"zkhost": AIPS_ZK_HOST, "collection": self.name,
-                "gen_uniq_key": "true", "commit_within": "5000"}
-        spark.sql(query).write.format("solr").options(**opts).mode("overwrite").save()
-        self.commit()
-    
     def add_documents(self, docs, commit=True):
         print(f"\nAdding Documents to '{self.name}' collection")
         response = requests.post(f"{SOLR_URL}/{self.name}/update?commit=true", json=docs).json()
```

Comment:
See my comment on [docker/data-science/notebooks/ch10/1.setup-the-movie-db.ipynb](https://github.com/treygrainger/ai-powered-search/pull/166/files#diff-a26d63a649c9507620ff58e769196f97a4087140548cbeea03ab16ab235b2c0e)]

Instead of of return the Solr-specific response, I'd suggest that we instead return Boolean indicating whether the add succeeded for failed.